dcp/DcpHandler.go

```diff
@@ -260,15 +260,15 @@ func (dh *DcpHandler) SnapshotMarker(snapshot gocbcore.DcpSnapshotMarker) {
 }
 
 func (dh *DcpHandler) Mutation(mutation gocbcore.DcpMutation) {
-	dh.writeToDataChan(CreateMutation(mutation.VbID, mutation.Key, mutation.SeqNo, mutation.RevNo, mutation.Cas, mutation.Flags, mutation.Expiry, gomemcached.UPR_MUTATION, mutation.Value, mutation.Datatype, mutation.CollectionID))
+	dh.writeToDataChan(CreateMutation(mutation.VbID, mutation.Key, mutation.SeqNo, mutation.RevNo, mutation.Cas, mutation.Flags, mutation.Expiry, gomemcached.UPR_MUTATION, mutation.Value, mutation.Datatype, mutation.CollectionID, dh.xattraIterator, dh.dcpClient.dcpDriver.xattrKeysForNoCompare))
```

Comment:
`writeToDataChan` is a method of `dh`. `dh` itself has access to `dh.xattriterator` and `dh.dcpClient`. No need to re-add it to function signature

Also, xattrIterator is mis-spelled

Reply:
"Simlarly, is there a way to not need to pass in arguments as part of Serialize()? mut can contain these from constructor, no?"
From this comment, I have added the iterator and the map containing keys for noCompare as a part of the mut constructor.
In the above line of code i am add them to the constructor of the "mutation struct".
writeToDataChan func signature remains the same